### PR TITLE
EG-1385 Exclusive sequence component accessibility fix part the first

### PR DESCRIPTION
**Pull request by woollett-r**

### CODE REVIEW

File: app/core/models/ocelot/package.scala [hunk 1]
```diff
@@ -42,7 +42,8 @@ package object ocelot {
   val listOfintegerRegex: Regex = s"$commaSeparatedIntsPattern".r
   val anyIntegerRegex: Regex = "^[\\-]?\\d+$".r
   val EmbeddedParameterRegex: Regex = """\{(\d)\}""".r
-  val exclusiveOptionRegex: Regex = """\[exclusive\]""".r
+  val exclusiveOptionRegex: Regex = "\\[exclusive:([A-za-z0-9:,\\s\\-_]+)\\]".r
```
Comment (cmrhmrc) [inline app/core/models/ocelot/package.scala:45]: Duplicate "z", need an A-Za-z here

File: app/models/ui/Sequence.scala [hunk 2]
```diff
@@ -27,6 +27,7 @@ case class ExclusiveSequence(text: Text,
                              hint: Option[Text],
                              options: Seq[Text],
                              exclusiveOption: Text,
+                             exclusiveOptionHint: Option[Text],
```
Comment (cmrhmrc) [inline app/models/ui/Sequence.scala:30]: Does accessibility allow this to be optional??

File: app/services/UIBuilder.scala [hunk 3]
```diff
@@ -244,12 +244,23 @@ class UIBuilder {
     val (text, hint) = TextBuilder.fromPhraseWithOptionalHint(exclusiveSequence.text)
     val options: Seq[Text] = exclusiveSequence.nonExclusiveOptions.map{phrase => TextBuilder.fromPhrase(phrase)}
 
+    val exclusiveOptionHintEnglish: Option[String] =
+      exclusiveOptionHintRegex.findAllMatchIn(exclusiveSequence.exclusiveOptions.head.english).toList.headOption.map(m => m.group(1))
```
Comment (cmrhmrc) [inline app/services/UIBuilder.scala:248]: can use findFirstMatchIn() as only one allowed

File: app/services/UIBuilder.scala [hunk 4]
```diff
@@ -244,12 +244,23 @@ class UIBuilder {
...
+
+    val exclusiveOptionHintWelsh: Option[String] =
+      exclusiveOptionHintRegex.findAllMatchIn(exclusiveSequence.exclusiveOptions.head.welsh).toList.headOption.map(m => m.group(1))
```
Comment (cmrhmrc) [inline app/services/UIBuilder.scala:251]: as above

File: app/core/models/ocelot/package.scala [hunk 5]
```diff
@@ -42,7 +42,8 @@ package object ocelot {
...
+  val exclusiveOptionHintRegex: Regex = "\\[exclusive:([^\\]]+)\\]".r
```
Comment (cmrhmrc) [inline app/core/models/ocelot/package.scala:46]: think you only need a single regex here, the latter which allows any non ] char would work

File: app/core/models/ocelot/package.scala (hunk 1 above)
Comment (woollett-r) [reply app/core/models/ocelot/package.scala:45]: Regex no longer used

File: app/core/models/ocelot/package.scala (hunk 5 above)
Comment (woollett-r) [reply app/core/models/ocelot/package.scala:46]: Done

File: app/models/ui/Sequence.scala (hunk 2 above)
Comment (woollett-r) [reply app/models/ui/Sequence.scala:30]: The optionality arises from the use of regex matching. In practice, owing to the mechanism used to identify exclusive sequences the hint will always be defined. I have kept the optionality as it provides a useful way of distinguishing between exclusive and non-exclusive sequences when rendering the sequence checkboxes.

File: app/services/UIBuilder.scala (hunk 3 above)
Comment (woollett-r) [reply app/services/UIBuilder.scala:248]: I've implemented findFirstMatchIn.

File: app/services/UIBuilder.scala (hunk 4 above)
Comment (woollett-r) [reply app/services/UIBuilder.scala:251]: As above

**Pull request merged**